control-plane-operator/controllers/hostedcontrolplane/pki/kas.go

```diff
@@ -129,9 +131,13 @@ func generateKubeConfig(url string, crtBytes, keyBytes, caBytes []byte) ([]byte,
 		Kind:       "Config",
 		APIVersion: "v1",
 	}
+	finalUrl, err := addBracketsIfIPv6(url)
```

Comment:
can we add a comment on why this step is needed? I'd expect the url coming through here to be valid already

Reply:
Yep, I will explain it here and also add a new comment on the code.

Basically, in the HostedCluster Spec we can specify how to expose the services. Even if it's a NodePort or LoadBalancer, the user can specify an IP or an URL. There is not any issues with URLs and IPv4s but the IPv6s needs additional brackets `https://[fd02::1]:8456`, if those brackets are not there, the Kubeconfig will not be functional and the `oc` client will return an error similar to:

```
couldn't get current server API group list: Get "https://fd2e:6f44:5dd8::58:31765/api?timeout=32s"
```

Because it's not prepared to handle IPv6 addresses without brackets and this confuses the ":" of the address with the port

Reply:
Thanks! I was just wondering if that'd be also a bug on the client but it seems legit to require the brackets in the url based on https://www.rfc-editor.org/rfc/rfc5952 and https://www.rfc-editor.org/rfc/rfc3986

Reply:
we only need the brackets for IP literal addresses not DNS hostnames, right?

Reply:
@muraee Yes, but only in IPv6 cases.